api/slug.go

```diff
@@ -12,13 +12,25 @@ func IssueSlug(issue *Issue) string {
 }
 
 // ProjectSlug is the slug formatter for projects.
-func ProjectSlug(project *Project) string {
-	return fmt.Sprintf("%s-%d", slug.Make(project.Name), project.ID)
+func ProjectSlug(project interface{}) string {
```

Comment:
I may choose to create ProjectPlainSlug and ProjectSlug respectively to leverage the compiler check

Reply:
@tianzhou I've considered that, but it seems to expose unnecessary implementation details: the caller does not care whether it's a server or store data model.

But put it another way, these are indeed two different functions which *happens to have almost the same implementation*. I'll change it to `ProjectPlainSlug` and `ProjectSlug` respectively.

PS: maybe we should take a look at the golang generics XD.

Reply:
:-) no generics plz

Reply:
Fixed.